sql/core/src/main/scala/org/apache/spark/sql/sources/DataSourceStrategy.scala

```diff
@@ -176,6 +176,9 @@ private[sql] object DataSourceStrategy extends Strategy {
       case expressions.Contains(a: Attribute, Literal(v: String, StringType)) =>
         Some(sources.StringContains(a.name, v))
 
+      case p: expressions.Predicate =>
```

Comment:
I don't think it's right to throw exception here, just in case people implement its own expression by inheriting from the `Predicate`.